src/include/mallocMC/mallocMC_hostclass.hpp

```diff
@@ -170,6 +178,13 @@ namespace mallocMC{
         return getAvailSlotsPoly(slotSize, boost::mpl::bool_<CreationPolicy::providesAvailableSlots::value>());
       }
 
+      MAMC_HOST
+      std::vector<mallocMC::HeapInfo> getHeapLocations(){
```

Comment:
You need to predefine `std::vector<HeapInfo>` in the top of the class else you get errors like this:

```
[ 16%] Building NVCC (Device) object build_picongpu/CMakeFiles/picongpu.dir//./picongpu_generated_main.cu.o
/home/widera/lib/mallocMC/include/mallocMC/mallocMC_hostclass.hpp(182): error: qualified name is not allowed

/home/widera/lib/mallocMC/include/mallocMC/mallocMC_hostclass.hpp(182): error: this declaration has no storage class or type specifier

/home/widera/lib/mallocMC/include/mallocMC/mallocMC_hostclass.hpp(181): error: attribute "host" does not apply here

/home/widera/lib/mallocMC/include/mallocMC/mallocMC_hostclass.hpp(182): error: expected a ";"

/home/widera/lib/mallocMC/include/mallocMC/mallocMC_hostclass.hpp(204): warning: parsing restarts here after previous syntax error

/bigdata/hplsim/scratch/widera/dev/src/picongpu/main.cu(74): error: class "mallocMC::Allocator<mallocMC::CreationPolicies::Scatter<ScatterConfig, mallocMC::CreationPolicies::ScatterConf::DefaultScatterHashingParams>, mallocMC::DistributionPolicies::Noop, mallocMC::OOMPolicies::ReturnNull, mallocMC::ReservePoolPolicies::SimpleCudaMalloc, mallocMC::AlignmentPolicies::Shrink<mallocMC::AlignmentPolicies::ShrinkConfig::DefaultShrinkConfig>>" has no member "getHeapLocations"
```

Define `typedef std::vector<HeapInfo> HeapInfoVector;` in the top of the class helps to avoid compile time errors. Errors are only triggert if `mallocMC:::getHeapLocations()` is used in the code.